acquisitions/src/main/resources/thunderjet/mod-finance/features/ledger-fiscal-year-rollover-fail-resistance-when-duplicate-encumbrance.feature

```diff
@@ -0,0 +1,175 @@
+
+    * callonce login testAdmin
+    * def okapitokenAdmin = okapitoken
+    * callonce login testUser
+    * def okapitokenUser = okapitoken
+
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenUser)', 'Accept': 'application/json' }
+    * def headersAdmin = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapitokenAdmin)', 'Accept': 'application/json' }
+
+    * configure headers = headersUser
+
+    * callonce variables
+
+    * def createFiscalYear = read('classpath:thunderjet/mod-finance/reusable/createFiscalYear.feature')
+    * def createLedger = read('classpath:thunderjet/mod-finance/reusable/createLedger.feature')
+    * def createOrder = read('classpath:thunderjet/mod-orders/reusable/create-order.feature')
+    * def createOrderLine = read('classpath:thunderjet/mod-orders/reusable/create-order-line.feature')
+    * def openOrder = read('classpath:thunderjet/mod-orders/reusable/open-order.feature')
+
+    * def fromFiscalYearId = callonce uuid1
+    * def toFiscalYearId = callonce uuid2
+    * def ledgerId = callonce uuid3
+    * def rolloverId = callonce uuid4
+    * def fundId = callonce uuid5
+    * def budgetId = callonce uuid6
+    * def orderId = callonce uuid7
+    * def poLineId = callonce uuid8
+
+    * def codePrefix = callonce random_string
+    * def fromYear = callonce getCurrentYear
+    * def toYear = parseInt(fromYear) + 1
+
+    * configure retry = { count: 10, interval: 5000 }
+
+  Scenario Outline: Prepare fiscal year with <fiscalYearId> for rollover
+    * configure headers = headersAdmin
+    * def fiscalYearId = <fiscalYearId>
+    * def code = <code>
+    * def periodStart = code + '-01-01T00:00:00Z'
+    * def periodEnd = code + '-12-30T23:59:59Z'
+
+    * def v = call createFiscalYear { id: #(fiscalYearId), code: #(codePrefix + code), periodStart: #(periodStart), periodEnd: #(periodEnd) }
+
+    Examples:
+      | fiscalYearId     | code     |
+      | fromFiscalYearId | fromYear |
+      | toFiscalYearId   | toYear   |
+
+
+  Scenario: Prepare finances
+    * configure headers = headersAdmin
+    * def v = call createLedger { id: #(ledgerId), fiscalYearId: #(fromFiscalYearId) }
+    * def v = call createFund { id: #(fundId),  ledgerId: #(ledgerId) }
+    * def v = call createBudget { id: #(budgetId), fundId: #(fundId), fiscalYearId: #(fromFiscalYearId), allocated: 100 }
+
+
+  Scenario: Create an order
+    * configure headers = headersAdmin
+    * def v = callonce createOrder { id: #(orderId), orderType: 'One-Time', reEncumber: true }
+
+
+  Scenario: Create an order line
+    * configure headers = headersAdmin
+    * def v = call createOrderLine { id: #(poLineId), orderId: #(orderId), fundId: #(fundId) }
+
+
+  Scenario: Open the order
+    * configure headers = headersAdmin
+    * def v = callonce openOrder { orderId: "#(orderId)" }
+
+
+  Scenario: Create a duplicate encumbrance
+    Given path 'finance-storage/order-transaction-summaries'
```

Comment:
Please don't use this deprecated API, it will be removed in Ramsons.

Reply:
Done